tests/test_tileserver.py

```diff
@@ -1,35 +1,79 @@
 """Tests for tileserver."""
 import pytest
+from click.testing import CliRunner
+from pathlib import Path
+import numpy as np
 
+from tiatoolbox import cli
 from tiatoolbox.visualization.tileserver import TileServer
 from tiatoolbox.wsicore.wsireader import WSIReader
 
 
 @pytest.fixture()
-def app(remote_sample) -> TileServer:
+def app(sample_ndpi) -> TileServer:
     """Create a testing TileServer WSGI app."""
-    wsi = WSIReader.open(remote_sample("ndpi-1"))
-    app = TileServer(title="test", layers={"ndpi": wsi})
+    runner = CliRunner()
+
+    # make a low-res .jpg of the right shape to be used as
+    # a low-res overlay
+    runner.invoke(
+        cli.main,
+        [
+            "slide-thumbnail",
+            "--img-input",
+            str(Path(sample_ndpi)),
+            "--mode",
+            "save",
+        ],
+    )
+    wsi = WSIReader.open(Path(sample_ndpi))
```

Comment:
why do you need this?

Reply:
Hi Shan,
I use it in line 40: `np.zeros(wsi.slide_dimensions(1.25, "power"), dtype=np.uint8).T` to make an array of suitable dimensions to test the case where a low-res heatmap is given as an np array for one of the layers